### ISSUE TITLE
Add support of Model Context Protocol (MCP)

### ISSUE THREAD

**Issue by ajshedivy** [i-am-bee/bee-agent-framework#238]
Original Issue: https://github.com/i-am-bee/bee-stack/issues/59

MCP was recently open-sourced by Anthropic as a standard way for LLM applications to interact with external datasources. 
- [MCP docs](https://modelcontextprotocol.io/introduction)
- [Github](https://github.com/modelcontextprotocol/)

MCP offers additional compatibility for developing Agents and LLM tools (like database connectors)

**Describe the solution you'd like**
Allow Bee agents to integrate with external Tools via MCP. 

**Describe alternatives you've considered**
Bee already has great ways to interact with external tools, I built a sample agent that can interact with Db2 for i: https://github.com/ajshedivy/bee-agent-ibmi

In Bee stack I can simulate MCP via REST API. My workaround at the moment is to create a simple webserver on IBM i that can talk to Db2 that accepts requests like:

```bash
curl -X POST http://HOST:PORT/api/execute-sql \
     -H "Content-Type: application/json" \
     -d '{"sql": "SELECT * FROM sample.employee"}'
```
This will return the result set from the given SQL. 
 
I can then create a "proxy tool" for Bee that sends requests to this endpoint:

```python
import requests

def send_sql_query(sql_query) -> dict:
    """
    Sends a SQL statement to the Flask server and returns the result.

    Args:
        sql_query (str): The SQL statement to execute.

    Returns:
        dict: The server's response as a dictionary.
    """
    url = "http://HOST:PORT/api/execute-sql"  # Server endpoint
    headers = {"Content-Type": "application/json"}
    payload = {"sql": sql_query}

    try:
        response = requests.post(url, json=payload, headers=headers)
        response.raise_for_status() 
        return response.json()     
    except requests.exceptions.RequestException as e:
        return {"error": str(e)}

```

This works decent:
![image](https://github.com/user-attachments/assets/654d3781-9def-4432-b3fa-f13d439d353f)

**Additional context**
In summary, MCP gives developers the ability to maintain a single, standardized external tool that would allow for additional flexibility when it comes to creating Agents and LLM apps. This would be a great addition to the Bee framework 😁

**Comment by pilartomas** [comment]
Hi @ajshedivy thanks for the issue, we're looking into it. 

We're thinking how to scope the initial implementation, would support for [resources](https://modelcontextprotocol.io/docs/concepts/resources) be sufficient for you at this point? Meaning [tools](https://modelcontextprotocol.io/docs/concepts/tools) would not be included.
